adhoc-openshift-pv.yml

```diff
@@ -0,0 +1,57 @@
+---
+# This playbook is to create a persistent volume for openshift CI namespace
+# Example to run the playbook
+# `ansible-playbook -e "env_prefix=ocp-prod" -e "pv_size=10Gi" -e
+# "cico_project_name=new_fancy_project" adhoc-openshift-pv.yml -i inventory`
+
+
+- hosts: ocp-ci-management
+  become: true
+
+  tasks:
+    - name: Generate a UUID
+      set_fact:
+        pv_uuid: "{{ (cico_project_name + pv_claimref|default('noclaimref', true)) | to_uuid }}"
+
+    - name: UUID Generated
+      debug:
+        var: pv_uuid
+
+    - name: Make a pv name
+      set_fact:
+        pv_name: "pv-{{ pv_size | lower }}-{{ pv_uuid }}"
+
+    - name: UUID Generated
+      debug:
+        var: pv_name
+
+    - name: See if the PV already exists
+      command: "oc get pv/{{ pv_name }}"
+      register: results
+      changed_when: false
+      ignore_errors: true
+
+    - name: Fail - the PV already exists
+      fail:
+        msg: "PV Already exists! {{ pv_name }}"
+      when:
+        - results.rc == 0
+        - not (force|default(false)|bool)
+
+    - name: make directories for each PV
+      file:
+        path: "/exports/{{ env_prefix }}/{{ pv_name }}"
+        owner: nfsnobody
+        group: nfsnobody
+        mode: 0777
+        state: directory
+      delegate_to: nfs02.ci.centos.org
```

Comment:
Never use any hardcoded hostname, as that goes against the "reusable" code logic, so that would need to be a variable, that it self is declared through group_vars for openshift